packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+
+### Background
+
+[Icon column type issue](https://github.com/ni/nimble/issues/1013)
+
+[Boolean column type issue](https://github.com/ni/nimble/issues/1103)
+
+### Features
+
+-   Supported input:
+    -   string
+    -   number
+    -   boolean
+-   Supported output:
+    -   Text
+    -   Icon
+    -   Spinner
+    -   (empty)
+
+### Non-goals
+
+-   Non-Nimble icon support
+-   Arbitrary icon colors
+-   Hyperlink icons
+-   Mixed text and icons
+-   Non-icon, non-spinner Nimble components
+
+---
+
+## Design
+
+Below is an example of how these elements would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-icon field-name="status" data-type="string">
+        Status
+        <nimble-mapping-icon key="fail" icon="nimble-icon-xmark" severity="error" label="Failed"></nimble-mapping-icon>
+        <nimble-mapping-icon key="error" icon="nimble-icon-xmark" severity="error" label="Errored"></nimble-mapping-icon>
+        <nimble-mapping-icon key="pass" icon="nimble-icon-check" severity="success" label="Passed"></nimble-mapping-icon>
+        <nimble-mapping-spinner key="running" label="Running"></nimble-mapping-spinner>
+    </nimble-table-column-icon>
+    <nimble-table-column-mapping field-name="errorCode" data-type="number">
+        Error Code
+        <nimble-mapping-text key="1" label="A bad thing happened"></nimble-mapping-text>
+        <nimble-mapping-text key="2" label="A worse thing happened"></nimble-mapping-text>
+        <nimble-mapping-text key="3" label="A terrible thing happened"></nimble-mapping-text>
+    </nimble-table-column-mapping>
+    <nimble-table-column-icon field-name="archived" data-type="boolean">
+        Archived
+        <nimble-mapping-icon key="true" icon="nimble-icon-database" label="Archived"></nimble-mapping-icon>
+    </nimble-table-column-icon>
+</nimble-table>
+```
+
+Each column contains mapping elements that define what to render when the cell's value matches the given `key` value.
+
+When none of the given mappings match the record value for a cell, that cell will be empty. Alternatively, if one of the mappings has the `default` attribute, it will match when no other mappings have. This is equivalent to the `placeholder` configuration we provide on `nimble-table-column-text` and `nimble-table-column-anchor`.
+
+Validation will be performed to ensure each mapping's key value can be converted to the `data-type` of the column. If not, an error flag will be set on the column's validation object. Note that whenever an error flag is set on the column's validation object, a generic `invalidColumnConfiguration` flag is also set on the table, putting it in an invalid state as well.
```

Comment:
I think we should go with the name `key-type`, better associates for users that it maps to the key attribute below

Reply:
Ok